clarity/src/vm/database/clarity_db.rs

```diff
@@ -964,26 +985,33 @@ impl<'a> ClarityDatabase<'a> {
     /// `get_current_block_height`).
     pub fn get_current_burnchain_block_height(&mut self) -> Result<u32> {
         let cur_stacks_height = self.store.get_current_block_height();
-        let last_mined_bhh = if cur_stacks_height == 0 {
-            return Ok(self.burn_state_db.get_burn_start_height());
-        } else {
-            self.get_index_block_header_hash(cur_stacks_height.checked_sub(1).ok_or_else(
-                || {
-                    InterpreterError::Expect(
-                        "BUG: cannot eval burn-block-height in boot code".into(),
-                    )
-                },
-            )?)?
-        };
 
-        self.get_burnchain_block_height(&last_mined_bhh)
-            .ok_or_else(|| {
-                InterpreterError::Expect(format!(
-                    "Block header hash '{}' must return for provided stacks block height {}",
-                    &last_mined_bhh, cur_stacks_height
-                ))
-                .into()
-            })
+        // In epoch 2, we can only access the burn block associated with the last block
+        if self.get_clarity_epoch_version()? < StacksEpochId::Epoch30 {
+            let last_mined_bhh = if cur_stacks_height == 0 {
+                return Ok(self.burn_state_db.get_burn_start_height());
+            } else {
+                // Safety note: normal subtraction is safe here, because we've already checked
+                // that cur_stacks_height > 0.
+                self.get_index_block_header_hash(cur_stacks_height - 1)?
+            };
```

Comment:
Would be more clear if you pull the `if...return` out of the `let` statement

```suggestion
            if cur_stacks_height == 0 {
                return Ok(self.burn_state_db.get_burn_start_height());
            };
            // Safety note: normal subtraction is safe here, because we've already checked
            // that cur_stacks_height > 0.
            let last_mined_bhh = self.get_index_block_header_hash(cur_stacks_height - 1)?;
```